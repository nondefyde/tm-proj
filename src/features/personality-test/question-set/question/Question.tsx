import React, { useState } from 'react';
import { AnswerType, QuestionType, QuestionTypeItem } from './index';
import { QuestionStyledButton, QuestionStyledContainer, QuestionStyledTitle, StyledButtons } from './Question.styled';
import { QuestionItem } from '../question-item';

export type QuestionProps = {
  currentIndex: number,
  answer?: AnswerType,
  question: QuestionType,
  onPrevious: (answer?: AnswerType) => void
  onNext: (answer?: AnswerType) => void
  answerQuestion: (answer: AnswerType) => void
}

export function Question(props: QuestionProps) {
  const { question, answer, answerQuestion, onPrevious, onNext, currentIndex } = props;
  
  const [selected, setSelected] = useState<AnswerType | undefined>(answer);
  
  console.log('selected ::: ', selected);
  
  const onSelected = (answer: AnswerType) => {
    setSelected(answer);
    answerQuestion(answer)
  }
  
  return (
    <QuestionStyledContainer>
      <QuestionStyledTitle>
        {question.title}
      </QuestionStyledTitle>
      {question.items.map((item: QuestionTypeItem) => {
        return <QuestionItem
          selectedAnswer={selected}
          questionId={question.id}
          onItemClicked={onSelected}
          key={item.id} item={item} />
      })}
      <StyledButtons>
        {currentIndex > 0 &&
          <QuestionStyledButton
            data-testid={'previous-button'}
            disabled={currentIndex <= 0}
            $isDefault={true}
            onClick={() => onPrevious(selected)}>
            Previous Question
          </QuestionStyledButton>
        }
        <QuestionStyledButton
          data-testid={'next-button'}
          disabled={!selected}
          onClick={() => onNext(selected)}>
          Next Question
        </QuestionStyledButton>
      </StyledButtons>
    </QuestionStyledContainer>
  );
}
