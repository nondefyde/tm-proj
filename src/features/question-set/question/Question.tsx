import React  from 'react';
import { QuestionType, QuestionTypeItem } from './index';
import { QuestionStyledButton, QuestionStyledContainer, QuestionStyledTitle } from './Question.styled';
import { QuestionItem } from '../question-item';

export type QuestionProps = {
  question: QuestionType,
}

export function Question(props: QuestionProps) {
  const {question} = props;
  
  const onNextButtonClicked = () => {}
  
  const onItemClicked = (item: QuestionTypeItem) => {
    console.log('OUTPUT item ::: ', item);
  }
  
  return (
    <QuestionStyledContainer>
      <QuestionStyledTitle>
        {question.title}
      </QuestionStyledTitle>
      {question.items.map((item: QuestionTypeItem) => {
        return <QuestionItem
          onItemClicked={onItemClicked}
          key={item.id} item={item} />
      })}
      <QuestionStyledButton onClick={onNextButtonClicked}>
        Next Question
      </QuestionStyledButton>
    </QuestionStyledContainer>
  );
}
