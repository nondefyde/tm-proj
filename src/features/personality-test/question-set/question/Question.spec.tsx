import { cleanup, render } from '@testing-library/react'
import { Question, QuestionProps } from './Question';
import { QuestionEnumType } from './index';

describe('Question Component', () => {
  
  afterEach(cleanup);
  
  let props: QuestionProps;
  
  beforeEach(() => {
    props = {
      currentIndex: -1,
      onNext: jest.fn(),
      onPrevious: jest.fn(),
      answerQuestion: jest.fn(),
      question: {
        id: '1',
        type: QuestionEnumType.MULTIPLE,
        title: 'Title',
        items: [
          {
            id: '1',
            identifier: 'A',
            description: 'Donâ€™t dare to interrupt them'
          }
        ]
      }
    };
  })
  
  it('should test snapshot', () => {
    const container = render(<Question {...props}/>)
    expect(container).toMatchSnapshot();
  });
});
