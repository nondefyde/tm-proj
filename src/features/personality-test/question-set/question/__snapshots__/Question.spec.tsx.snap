// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Question Component should test snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-bcXHqe iZPgNH"
      >
        <div
          class="sc-gswNZR jxTqzW"
        >
          Title
        </div>
        <div
          class="sc-eDvSVe gsZhvf"
        >
          <div
            class="sc-jSUZER hhsxcT"
          >
            <span>
              A
            </span>
          </div>
          <div
            class="sc-gKPRtg kuQAMJ"
          >
            Don’t dare to interrupt them
          </div>
        </div>
        <div
          class="sc-hLBbgP bXfvrS"
        >
          <button
            class="sc-dkrFOg fvHCOW"
            disabled=""
          >
            Next Question
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bcXHqe iZPgNH"
    >
      <div
        class="sc-gswNZR jxTqzW"
      >
        Title
      </div>
      <div
        class="sc-eDvSVe gsZhvf"
      >
        <div
          class="sc-jSUZER hhsxcT"
        >
          <span>
            A
          </span>
        </div>
        <div
          class="sc-gKPRtg kuQAMJ"
        >
          Don’t dare to interrupt them
        </div>
      </div>
      <div
        class="sc-hLBbgP bXfvrS"
      >
        <button
          class="sc-dkrFOg fvHCOW"
          disabled=""
        >
          Next Question
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
